src/dealDashboard/dealDiscussion/discussionThread/discussionThread.ts

```diff
@@ -381,27 +366,27 @@ export class DiscussionThread {
 
     const swrVote = Utils.cloneDeep(this.threadDictionary[_id]);
 
-    this.discussionsService.voteComment(this.discussionId, _id, type).then(res => {
-      if (res.error) {
-        this.eventAggregator.publish("handleFailure", "An error occurred while voting. " + res.error);
-        if (res.code === 404) {
-          delete this.threadDictionary[_id];
-          this.threadComments = [...Object.values(this.threadDictionary)];
+    this.discussionsService.voteComment(this.discussionId, _id, type)
+      .catch(error => {
+        /* On API failure- revert voting */
+        if (error.code === 404) {
+          if (error.error) {
+            this.eventAggregator.publish("handleFailure", "An error occurred while voting." + error.error);
+            this.threadComments = this.threadComments.filter(comment => comment._id !== _id);
+            this.threadDictionary = this.arrayToDictionary(this.threadComments);
```

Comment:
Only revert vote action, when api error occured.
In this case, it was very likely, that the comment was deleted by the original author (note error code 404)